pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -2654,7 +2654,7 @@ func assertState(s state) step {
 				}
 				if len(s.absentOwnerCache) != ctx.gc.absentOwnerCache.cache.Len() {
 					// only way to inspect is to drain them all, but that's ok because we're failing the test anyway
-					ctx.gc.absentOwnerCache.cache.OnEvicted = func(key lru.Key, item interface{}) {
+					err := ctx.gc.absentOwnerCache.cache.SetEvictionFunc(func(key lru.Key, item interface{}) {
```

Comment:
only needed for a test?

Reply:
yep

Reply:
🤷 🙃

Reply:
I still think this can be racy without a lock in SetEvictionFunc, but in this specific use, it doesn't really matter (this is helping debug an already-failed test). I do think a follow-up in utils to make SetEvictionFunc threadsafe would be good